docs/Client.md

```diff
@@ -0,0 +1,1011 @@
+
+As demonstrated in [Example 1 - Basic GET stream request](#example-1---basic-get-stream-request), it is recommended to use the `option.opaque` property to avoid creating a closure for the `factory` method. This pattern works well with Node.js Web Frameworks such as [Fastify](https://fastify.io). See [Example 2 - Stream to Fastify Response](#example-2---stream-to-fastify-response) for more details.
+
+Arguments:
+
+* **options** `RequestOptions`
+* **factory** `(data: StreamFactoryData) => stream.Writable`
+* **callback** `(error: Error | null, data: StreamData) => void` (optional)
+
+Returns: `void | Promise<StreamData>` - Only returns a `Promise` if no `callback` argument was passed
+
+#### Parameter: `StreamFactoryData`
+
+* **statusCode** `number`
+* **headers** `http.IncomingHttpHeaders`
+* **opaque** `unknown`
+
+#### Parameter: `StreamData`
+
+* **opaque** `unknown`
+* **trailers** `Record<string, string>`
+
+#### Example 1 - Basic GET stream request
+
+```js
+'use strict'
+const { createServer } = require('http')
+const { Client } = require('undici')
+const { Writable } = require('stream')
+
+const server = createServer((request, response) => {
+  response.end('Hello, World!')
+})
+
+server.listen(() => {
+  const client = new Client(`http://localhost:${server.address().port}`)
+
+  const bufs = []
+
+  client.stream({
+    path: '/',
+    method: 'GET',
+    opaque: { bufs }
+  }, ({ statusCode, headers, opaque: { bufs } }) => {
+    console.log(`response received ${statusCode}`)
+    console.log('headers', headers)
+    return new Writable({
+      write (chunk, encoding, callback) {
+        bufs.push(chunk)
+        callback()
+      }
+    })
+  }).then(({ opaque: { bufs } }) => {
+    console.log(Buffer.concat(bufs).toString('utf-8'))
+
+    client.close()
+    server.close()
+  }).catch(error => {
+    console.error(error)
+  })
+})
+```
+
+#### Example 2 - Stream to Fastify Response
+
+In this example, a (fake) request is made to the fastify server using `fastify.inject()`. This request then executes the fastify route handler which makes a subsequent request to the raw Node.js http server using `undici.client.stream()`. The fastify response is passed to the `opaque` option so that undici can tap into the underlying writable stream using `response.raw`. This methodology demonstrates how one could use undici and fastify together to create fast-as-possible requests from one backend server to another.
+
+```js
+'use strict'
+
+const { createServer } = require('http')
+const undici = require('undici')
+const fastify = require('fastify')
+
+const nodeServer = createServer((request, response) => {
+  response.end('Hello, World! From Node.js HTTP Server')
+})
+
+nodeServer.listen(() => {
+  console.log('Node Server listening')
+
+  const nodeServerUndiciClient = new undici.Client(`http://localhost:${nodeServer.address().port}`)
```

Comment:
Should we use static or instance methods in examples like this? I think static makes it easier.

Reply:
I think I answered this previously, but I'll reiterate anyways. I'd like to keep these embedded examples as close to the relative method so that users can see how they are used independently. For "best way to use undici" we should have a separate doc page/section.